test/cartesian_analysis_test.rb

```diff
@@ -6,79 +6,165 @@
 #
 # -----------------------------------------------------------------------------
 
-require "test_helper"
+require_relative "test_helper"
 
 class CartesianAnalysisTest < Minitest::Test # :nodoc:
+  class Fixtures
+    def initialize(factory)
+      @factory = factory
+    end
+
+    def clockwise_triangle
+      p1 = @factory.point(1, 1)
+      p2 = @factory.point(2, 4)
+      p3 = @factory.point(5, 2)
+      @factory.line_string([p1, p2, p3, p1])
+    end
+
+    def counterclockwise_triangle
+      p1 = @factory.point(1, 1)
+      p2 = @factory.point(2, 4)
+      p3 = @factory.point(5, 2)
+      @factory.line_string([p1, p3, p2, p1])
+    end
+
+    def clockwise_puckered_quad
+      p1 = @factory.point(1, 1)
+      p2 = @factory.point(2, 6)
+      p3 = @factory.point(3, 3)
+      p4 = @factory.point(5, 2)
+      @factory.line_string([p1, p2, p3, p4, p1])
+    end
+
+    def counterclockwise_puckered_quad
+      p1 = @factory.point(1, 1)
+      p2 = @factory.point(2, 6)
+      p3 = @factory.point(3, 3)
+      p4 = @factory.point(5, 2)
+      @factory.line_string([p1, p4, p3, p2, p1])
+    end
+
+    def clockwise_hat
+      p1 = @factory.point(1, 2)
+      p2 = @factory.point(2, 3)
+      p3 = @factory.point(3, 2)
+      p4 = @factory.point(2, 1)
+      p5 = @factory.point(2, 0)
+      p6 = @factory.point(0, 2)
+      @factory.line_string([p1, p2, p3, p4, p5, p6, p1])
+    end
+
+    def counterclockwise_hat
+      p1 = @factory.point(2, 1)
+      p2 = @factory.point(3, 2)
+      p3 = @factory.point(2, 3)
+      p4 = @factory.point(1, 2)
+      p5 = @factory.point(0, 2)
+      p6 = @factory.point(2, 0)
+      @factory.line_string([p1, p2, p3, p4, p5, p6, p1])
+    end
+
+    def counterclockwise_near_circle
+      p1 = @factory.point(0, -3)
+      p2 = @factory.point(2, -2)
+      p3 = @factory.point(3, 0)
+      p4 = @factory.point(2, 2)
+      p5 = @factory.point(0, 3)
+      p6 = @factory.point(-2, 2)
+      p7 = @factory.point(-3, 0)
+      p8 = @factory.point(-2, -2)
+      @factory.line_string([p1, p2, p3, p4, p5, p6, p7, p8, p1])
+    end
+  end
+
   def setup
-    @factory = RGeo::Cartesian.simple_factory
+    @fixtures = Fixtures.new(RGeo::Cartesian.simple_factory)
+    @cfixtures = Fixtures.new(RGeo::Geos.factory) if RGeo::Geos.supported?
```

Comment:
If `RGeo::Geos.supported?` is `false`, all of the tests that use `@cfixtures` will fail.

Perhaps a better solution would be to include the `CartesianAnalysisTest` class in the `GeosCapi::AnalysisTest` module and run the fixtures with a `CAPIFactory`. We would also have to remove all of the combined/cfixture tests from here and just test `@factory`. The `@cfixture` specific tests would be moved to the c_api test module and `RGeo::Cartesian::Analysis.ring_direction` would be undefined as well so they are not run with a geos factory.

I can handle this if you think it makes sense to do.

Reply:
Good catch, moreover, the method here should be:
```suggestion
    @cfixtures = Fixtures.new(RGeo::Geos.factory) if RGeo::Geos.capi_supported?
```

However, we could just do something like:

```suggestion
    @cfixtures =
      if RGeo::Geos.capi_supported?
      	Fixtures.new(RGeo::Geos.factory(native_interface: :capi))
      else
        @fixtures
      end
```

Because the tests should not fait if capi is not supported, since there is a fallback. But we'll just test twice the same method in that case, hence your solution may be nice. However, there are two different methods: `ring_direction` and `ccw?`, and the capi is only interacting with the second one. The reason for that is that the GEOS method doesn't explicitly say that direction is clockwise when it returns false, it just says that it is not counter-clockwise.

Anyway, if you have the time to do it now I'm all for that :)

Or I'll do it next week